front_end/core/sdk/HeapProfilerModel.ts

```diff
@@ -106,8 +106,9 @@ export class HeapProfilerModel extends SDKModel<EventTypes> {
     return Boolean(response.getError());
   }
 
-  async takeHeapSnapshot(heapSnapshotOptions: Protocol.HeapProfiler.TakeHeapSnapshotRequest): Promise<void> {
-    await this.#heapProfilerAgent.invoke_takeHeapSnapshot(heapSnapshotOptions);
+  async takeHeapSnapshot(heapSnapshotOptions: Protocol.HeapProfiler.TakeHeapSnapshotRequest): Promise<boolean> {
+    const response = await this.#heapProfilerAgent.invoke_takeHeapSnapshot(heapSnapshotOptions);
+    return Boolean(response.getError());
```

Comment:
Rather than returning `true` for failed, can we return positively for a success?

Reply:
Was just talking about this with @hoxyq. `true`-for-success is my preference and intuition too. However, the existing CDT returns `true` for failure:

https://github.com/facebookexperimental/rn-chrome-devtools-frontend/blob/7f6cfe3c847411c3e4a730384d00536f049edc16/front_end/core/sdk/HeapProfilerModel.ts#L104-L122

In the existing usage, the return booleans are never used, i.e. all of them are discarded with `void`. So there's no material change if we flip the boolean.

If we decide that switching all existing call sites to `true`-for-success is worth the potential future merge conflicts, I'm happy to follow up in a separate PR for this.

Reply:
Thanks for the context. Won't block on this 👍🏻